controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    mecanum_drive_controller:
      type: mechai_controls/MecanumDriveController

mecanum_drive_controller:
  ros__parameters:
    # Wheel position names are base on looking top down with the x configuration of wheels
    wheel_names: ["front_left_wheel_joint", "front_right_wheel_joint", "back_left_wheel_joint", "back_right_wheel_joint"]

    wheel_separation_w: 0.4
    wheel_separation_l: 0.4
    wheel_radius: 0.1

    # publish_rate: 50.0
    # odom_frame_id: odom
    # base_frame_id: base_link
    # pose_covariance_diagonal : [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    # twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]

    open_loop: true
    # enable_odom_tf: false

    cmd_vel_timeout: 0.5
    #publish_limited_velocity: true
    # use_stamped_vel: false
    #velocity_rolling_window_size: 10

    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    # linear.x.has_velocity_limits: true
    # linear.x.has_acceleration_limits: true
    # linear.x.has_jerk_limits: false
    # linear.x.max_velocity: 1.0
    # linear.x.min_velocity: -1.0
    # linear.x.max_acceleration: 1.0
    # linear.x.max_jerk: 0.0
    # linear.x.min_jerk: 0.0

    # angular.z.has_velocity_limits: true
    # angular.z.has_acceleration_limits: true
    # angular.z.has_jerk_limits: false
    # angular.z.max_velocity: 1.0
    # angular.z.min_velocity: -1.0
    # angular.z.max_acceleration: 1.0
    # angular.z.min_acceleration: -1.0
    # angular.z.max_jerk: 0.0
    # angular.z.min_jerk: 0.0
